// mysql2/index.js
var Connection       = require('mysql2/lib/connection.js.js');
var ConnectionConfig = require('mysql2/lib/connection_config.js.js');
var SqlString        = require('mysql2/lib/sql_string.js.js');

module.exports.createConnection = function(opts) {
  return new Connection({config: new ConnectionConfig(opts)});
};

module.exports.connect = module.exports.createConnection;

module.exports.Connection = Connection;
module.exports.Types      = require('mysql2/lib/constants/types.js.js');

module.exports.createPool = function(config) {
  var PoolConfig = require('mysql2/lib/pool_config.js.js');
  var Pool       = require('mysql2/lib/pool.js.js');
  return new Pool({config: new PoolConfig(config)});
};

exports.createPoolCluster = function(config) {
  var PoolCluster = require('mysql2/lib/pool_cluster.js.js');
  return new PoolCluster(config);
};

module.exports.createServer = function() {
  var Server = require('mysql2/lib/server.js.js');
  return new Server();
};

exports.escape   = SqlString.escape;
exports.escapeId = SqlString.escapeId;
exports.format   = SqlString.format;
